src/deletepy/cli/main.py

```
@@ -127,6 +127,93 @@ def revoke_grants_only(input_file: Path, env: str, dry_run: bool) -> None:
+    "--dry-run", is_flag=True, help="Preview what would be cleaned without actually deleting"
+)
+def clean(clean_all: bool, failed: bool, days_old: int, dry_run: bool) -> None:
+    """Clean up old or failed checkpoints."""
+    handler = OperationHandler()
+    handler.handle_clean_checkpoints(clean_all, failed, days_old, dry_run)
+
+
+@checkpoint.command()
+@click.argument("checkpoint_id", type=str)
+@click.option("--confirm", is_flag=True, help="Skip confirmation prompt")
+def delete(checkpoint_id: str, confirm: bool) -> None:
```

Ruff (0.11.9)

Redefinition of unused `delete` from line 114

(F811)